src/Avalonia.Base/Threading/AvaloniaSynchronizationContext.cs

```diff
@@ -30,7 +30,7 @@ public static void InstallIfNeeded()
         /// <inheritdoc/>
         public override void Post(SendOrPostCallback d, object? state)
         {
-           Dispatcher.UIThread.Post(d, state, DispatcherPriority.Background);
+            Dispatcher.UIThread.Post(() => d(state), DispatcherPriority.Background);
```

Comment:
to not forget, this method should use overload that accepts state as a parameter instead of creating a closure